src/app/clusters/icd-management-server/icd-management-server.cpp

```diff
@@ -1,510 +0,0 @@
-/**
- *
- *    Copyright (c) 2023 Project CHIP Authors
- *
- *    Licensed under the Apache License, Version 2.0 (the "License");
- *    you may not use this file except in compliance with the License.
- *    You may obtain a copy of the License at
- *
- *        http://www.apache.org/licenses/LICENSE-2.0
- *
- *    Unless required by applicable law or agreed to in writing, software
- *    distributed under the License is distributed on an "AS IS" BASIS,
- *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- *    See the License for the specific language governing permissions and
- *    limitations under the License.
- */
-
-#include "icd-management-server.h"
-
-#include <access/AccessControl.h>
-#include <access/Privilege.h>
-#include <app-common/zap-generated/attributes/Accessors.h>
-#include <app-common/zap-generated/cluster-enums.h>
-#include <app-common/zap-generated/cluster-objects.h>
-#include <app-common/zap-generated/ids/Clusters.h>
-#include <app/AttributeAccessInterface.h>
-#include <app/AttributeAccessInterfaceRegistry.h>
-#if CHIP_CONFIG_ENABLE_ICD_CIP
-#include <app/icd/server/ICDNotifier.h> // nogncheck
-#endif
-#include <app/server/Server.h>
-#include <app/util/attribute-storage.h>
-
-using namespace chip;
-using namespace chip::app;
-using namespace chip::app::Clusters;
-using namespace chip::app::Clusters::IcdManagement;
-using namespace Protocols;
-using namespace chip::Access;
-
-using chip::Protocols::InteractionModel::Status;
-
-namespace {
-
-/**
- * @brief Implementation of attribute access for IcdManagement cluster
- */
-class IcdManagementAttributeAccess : public AttributeAccessInterface
-{
-public:
-    IcdManagementAttributeAccess() : AttributeAccessInterface(MakeOptional(kRootEndpointId), IcdManagement::Id) {}
-
-    void Init(PersistentStorageDelegate & storage, Crypto::SymmetricKeystore * symmetricKeystore, FabricTable & fabricTable,
-              ICDConfigurationData & icdConfigurationData)
-    {
-#if CHIP_CONFIG_ENABLE_ICD_CIP
-        mStorage           = &storage;
-        mSymmetricKeystore = symmetricKeystore;
-        mFabricTable       = &fabricTable;
-#endif // CHIP_CONFIG_ENABLE_ICD_CIP
-        mICDConfigurationData = &icdConfigurationData;
-    }
-
-    CHIP_ERROR Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder) override;
-
-private:
-    CHIP_ERROR ReadIdleModeDuration(EndpointId endpoint, AttributeValueEncoder & encoder);
-    CHIP_ERROR ReadActiveModeDuration(EndpointId endpoint, AttributeValueEncoder & encoder);
-    CHIP_ERROR ReadActiveModeThreshold(EndpointId endpoint, AttributeValueEncoder & encoder);
-    CHIP_ERROR ReadFeatureMap(EndpointId endpoint, AttributeValueEncoder & encoder)
-    {
-        return encoder.Encode(mICDConfigurationData->GetFeatureMap());
```

Comment:
This is breaking existing consumers that are setting the FeatureMap via the configuration data, no?

Why are we doing that?